src/NHibernate/Driver/SqlClientDriver.cs

```diff
@@ -97,13 +105,16 @@ public override bool SupportsMultipleOpenReaders
 		protected override void InitializeParameter(DbParameter dbParam, string name, SqlType sqlType)
 		{
 			base.InitializeParameter(dbParam, name, sqlType);
-			SetVariableLengthParameterSize(dbParam, sqlType);
+			if (_dialect == null)
+				throw new InvalidOperationException("Dialect not available, is this driver used without having been configured?");
+			SetVariableLengthParameterSize(dbParam, sqlType, _dialect.DefaultCastPrecision, _dialect.DefaultCastScale);
 		}
 
 		// Used from SqlServerCeDriver as well
-		public static void SetVariableLengthParameterSize(DbParameter dbParam, SqlType sqlType)
+		public static void SetVariableLengthParameterSize(
+			DbParameter dbParam, SqlType sqlType, byte defaultDecimalPrecision, byte defaultDecimalScale)
```

Comment:
If doing that in 5.1, I guess we will have to add back previous method but flagged obsolete, for avoiding a possible breaking change.